packages/svelte/src/compiler/phases/1-parse/read/style.js

```diff
@@ -187,42 +186,90 @@ function read_selector_list(parser, inside_pseudo_class = false) {
 function read_selector(parser, inside_pseudo_class = false) {
 	const list_start = parser.index;
 
-	/** @type {Array<import('#compiler').Css.SimpleSelector | import('#compiler').Css.Combinator>} */
+	/** @type {import('#compiler').Css.RelativeSelector[]} */
 	const children = [];
 
+	/**
+	 * @param {import('#compiler').Css.Combinator | null} combinator
+	 * @param {number} start
+	 * @returns {import('#compiler').Css.RelativeSelector}
+	 */
+	function create_selector(combinator, start) {
+		return {
+			type: 'RelativeSelector',
+			combinator,
+			selectors: [],
+			start,
+			end: -1,
+			metadata: {
+				is_global: false,
+				is_host: false,
+				is_root: false,
+				scoped: false
+			}
+		};
+	}
+
+	/** @type {import('#compiler').Css.RelativeSelector} */
+	let relative_selector = create_selector(null, parser.index);
+
 	while (parser.index < parser.template.length) {
-		const start = parser.index;
+		let start = parser.index;
+
+		const combinator = read_combinator(parser);
```

Comment:
AFAIR this should come _after_ matching `nth` to avoid false positives where p:nth-of-type(+3n) would be considered a combinator

Reply:
ah yeah, you're right:

<img width="1310" alt="image" src="https://github.com/sveltejs/svelte/assets/1162160/ff782014-378f-4aca-8af7-dc7a66ab509b">

it's totally harmless, since in the places where this would be valid we don't actually care about the pseudo element arguments. but i will feel bad about the malformed AST, so it needs fixing

Reply:
fixed, thanks!